pkg/compose/down.go

```diff
@@ -148,32 +148,42 @@ func (s *composeService) ensureImagesDown(ctx context.Context, project *types.Pr
 
 func (s *composeService) ensureNetworksDown(ctx context.Context, project *types.Project, w progress.Writer) []downOp {
 	var ops []downOp
-	for _, n := range project.Networks {
+	for key, n := range project.Networks {
 		if n.External.External {
 			continue
 		}
 		// loop capture variable for op closure
+		networkKey := key
 		networkName := n.Name
 		ops = append(ops, func() error {
-			return s.removeNetwork(ctx, networkName, w)
+			return s.removeNetwork(ctx, networkKey, project.Name, networkName, w)
 		})
 	}
 	return ops
 }
 
-func (s *composeService) removeNetwork(ctx context.Context, name string, w progress.Writer) error {
+func (s *composeService) removeNetwork(ctx context.Context, networkName string, projectName string, name string, w progress.Writer) error {
 	// networks are guaranteed to have unique IDs but NOT names, so it's
 	// possible to get into a situation where a compose down will fail with
 	// an error along the lines of:
 	// 	failed to remove network test: Error response from daemon: network test is ambiguous (2 matches found based on name)
 	// as a workaround here, the delete is done by ID after doing a list using
 	// the name as a filter (99.9% of the time this will return a single result)
 	networks, err := s.apiClient().NetworkList(ctx, moby.NetworkListOptions{
-		Filters: filters.NewArgs(filters.Arg("name", name)),
+		Filters: filters.NewArgs(
+			projectFilter(projectName),
+			networkFilter(networkName),
+			filters.Arg("name", name)),
 	})
 	if err != nil {
-		return errors.Wrapf(err, fmt.Sprintf("failed to inspect network %s", name))
+		return errors.Wrapf(err, "failed to list networks")
 	}
+
+	// NetworkList API doesn't return the exact name match, so we can retrieve more than one network with a request
+	networks = utils.Filter(networks, func(net moby.NetworkResource) bool {
+		return net.Name == name
+	})
```

Comment:
This is unnecessary - `NetworkList` with a name filter works reliably, it's `NetworkInspect` that does prefix matching